extensions/ui-extension/src/fields/Newsletter.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+   Form,
+   Button,
+   BlockStack,
+   Heading,
+   TextField,
+   Text,
+   useTranslate,
+   Grid,
+   Banner,
+} from '@shopify/ui-extensions-react/checkout';
+import { useState } from 'react';
+
+const ifixit_base_url = 'https://www.ifixit.com';
+
+const emailRegex = /^([a-zA-Z0-9._%-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,})$/;
+
+function isValidEmail(email: string) {
+   return emailRegex.test(email);
+}
+
+export function Newsletter() {
+   const [showEntryBox, setShowEntryBox] = useState(true);
+   const [showError, setShowError] = useState(false);
+   const [validationError, setValidationError] = useState('');
+   const [isLoading, setShowLoading] = useState(false);
+   let email = '';
+
+   const subscribeEmail = (email: string) => {
+      if (!isValidEmail(email)) {
+         setValidationError('Please enter a valid email');
+         return;
+      }
+      setShowLoading(true);
+      fetch(`${ifixit_base_url}/api/2.0/cart/newsletter/subscribe`, {
+         method: 'POST',
+         body: JSON.stringify({ email: email }),
+      })
+         .then((response) => {
+            setShowEntryBox(false);
+            setShowLoading(false);
+            if (response.ok) {
+               setShowError(false);
+            } else {
+               console.error(response);
+               setShowError(true);
+            }
+         })
+         .catch((error) => {
+            console.error(error);
+            setShowEntryBox(false);
+            setShowError(true);
+         });
+   };
+
+   const translate = useTranslate();
+   const heading = translate('newsletter.heading');
+   const description = translate('newsletter.description');
+   const field_label = translate('newsletter.field_label');
+   const button_label = translate('newsletter.button_label');
+   const success_message = translate('newsletter.success_message');
+   const error_message = translate('newsletter.error_message');
+
+   return (
+      <Form onSubmit={() => subscribeEmail(email)}>
+         <BlockStack border="base" borderRadius="base" padding="base">
+            <Heading>{heading}</Heading>
+            <Text>{description}</Text>
+            {showEntryBox && (
+               <Grid columns={['80%', '20%']} spacing="base">
+                  <TextField
+                     label={field_label}
+                     onChange={(value) => (email = value)}
+                     error={validationError}
+                     onInput={() => setValidationError('')}
+                  />
```

Comment:
```suggestion
                  <TextField
                     label={field_label}
                     type="email"
                     onChange={(value) => (email = value)}
                     error={validationError}
                     onInput={() => setValidationError('')}
                  />
```
https://shopify.dev/docs/api/checkout-ui-extensions/2023-10/components/forms/textfield#textfieldprops-propertydetail-type